mobile/src/helpers/health.ts

```diff
@@ -2,14 +2,138 @@ import GoogleFit, { Scopes } from 'react-native-google-fit';
 import AppleHealthKit from 'react-native-health';
 import logger from 'common/logger';
 
+// GOOGLE FIT:
 const runOptions = {
   scopes: [
     Scopes.FITNESS_ACTIVITY_READ,
-    Scopes.FITNESS_ACTIVITY_WRITE,
-    Scopes.FITNESS_BODY_READ,
-  ]
+    Scopes.FITNESS_SLEEP_READ
+  ],
 };
 
+//Google Fit Steps
+let dailyStepCount;
+
+var today = new Date();
+var lastWeekDate = new Date(
+  today.getFullYear(),
+  today.getMonth(),
+  today.getDate() - 8,
+);
+const opt = {
+  startDate: lastWeekDate.toISOString(), // required ISO8601Timestamp
+  endDate: today.toISOString(), // required ISO8601Timestamp
+  bucketUnit: 'DAY', // optional - default "DAY". Valid values: "NANOSECOND" | "MICROSECOND" | "MILLISECOND" | "SECOND" | "MINUTE" | "HOUR" | "DAY"
+  bucketInterval: 1, // optional - default 1.
+};
+
+const fetchStepsData = async opt => {
+  const res = await GoogleFit.getDailyStepCountSamples(opt);
+  if (res.length !== 0) {
+    for (var i = 0; i < res.length; i++) {
+      if (res[i].source === 'com.google.android.gms:estimated_steps') {
+        let data = res[i].steps.reverse();
+        dailyStepCount = res[i].steps;
+        // setdailySteps(data[0].value);
+        return dailyStepCount;
+      }
+    }
+  } else {
+    logger.log('Daily Steps Not Found');
+  }
+};
+
+const fetchSleepData = async opt => {
+  const midnight = "00:00"
+  // var midnight = new Date();
+  // midnight.setHours(0, 0, 0, 0);
+  let sleepTotal = 0;
+  const res = await GoogleFit.getSleepSamples(opt);
+
+  for (var i = 0; i < res.length; i++) {
```

Comment:
this is for loop is quite difficult to interpret, any way to simplify? A comment describing what the loop is doing would be helpful

Reply:
left comment